src/module_2/module_2_EDA.md

```diff
@@ -0,0 +1,2197 @@
+
+
+```python
+df_variant_user_counts = df_regulars.groupby('variant_id')['user_id'].nunique().reset_index()
+df_variant_user_counts.rename(columns={'user_id': 'user_id_count'}, inplace=True)
+df_inventory_merged = pd.merge(df_inventory, df_variant_user_counts, how='left', on='variant_id')
+```
+
+
+```python
+print(df_inventory_merged.isnull().sum())
+```
+
+    variant_id            0
+    price                 0
+    compare_at_price      0
+    vendor                0
+    product_type          0
+    tags                  0
+    user_id_count       448
+    dtype: int64
+
+
+Parece que hay una serie de varianted el dataframe de inventory las cuales no estaban en el dataframe de regulars. Vamos a comprobarlo rapidamente para ver que estamos haciendo todo bien:
+
+
+```python
+print(f'Valores únicos de inventory: {len(df_inventory['variant_id'].unique())}')
+print(f'Valores únicos de regulars: {len(df_regulars['variant_id'].unique())}')
+```
+
+    Valores únicos de inventory: 1733
+    Valores únicos de regulars: 1843
+
+
+
+```python
+unicos_inventory = df_inventory['variant_id'].unique()
+unicos_regulars = df_regulars['variant_id'].unique()
+valores_coincidentes = np.intersect1d(unicos_inventory, unicos_regulars)
+
+print(f'Valores únicos coincidentes entre inventory y regulars: {len(unicos_inventory) - len(valores_coincidentes)}')
+```
+
+    Valores únicos coincidentes entre inventory y regulars: 448
+
+
+**Estudio de la popularidad por marca:**
+
+
+```python
+df_inventory_merged['user_id_count'] = df_inventory_merged['user_id_count'].fillna(0).astype(int)
+
+df_inventory_merged_sorted = df_inventory_merged.sort_values(by='user_id_count', ascending=False)
+
+plt.figure(figsize=(6, 4))
+plt.bar(df_inventory_merged_sorted['vendor'].head(20), df_inventory_merged_sorted['user_id_count'].head(20), color='skyblue')
+plt.xlabel('Vendor')
+plt.ylabel('User ID Count')
+plt.title('Top 20 User ID Count by Vendor for Each Variant ID')
+plt.xticks(rotation=45, ha='right')
+plt.tight_layout()
+
+plt.show()
+```
+
+
+    
+![png](module_2_EDA_files/module_2_EDA_23_0.png)
+    
+
+
+
+```python
+df_vendor_popularity = df_inventory_merged_sorted.groupby('vendor')['user_id_count'].sum().sort_values(ascending=False)
+plt.figure(figsize=(6, 4))
+df_vendor_popularity.head(20).plot(kind='bar', color='skyblue')
+plt.xlabel('Vendor')
+plt.ylabel('Total User ID Count')
+plt.title('Top 20 Vendor Popularity by Total User ID Count')
+plt.xticks(rotation=45, ha='right')
+plt.tight_layout()
+
+plt.show()
+```
+
+
+    
+![png](module_2_EDA_files/module_2_EDA_24_0.png)
+    
+
+
+**Estudio de la popularidad por precio:**
+
+
+```python
+
+plt.figure(figsize=(6, 4))
+plt.barh(df_inventory_merged_sorted['vendor'].head(10), df_inventory_merged_sorted['price'].head(10), color='skyblue')
+plt.xlabel('Price')
+plt.ylabel('Vendor')
+plt.title('Top 10 Variant IDs by User ID Count with Corresponding Price')
```

Comment:
What is the exact purpose of this plot, what info is giving us? Also why the other ones were vertical and this is horizontal?